var/spack/repos/builtin/packages/rocfft/0005-Fix-clients-tests-include-rocrand-fftw-include-dir-rocm-6.0.0.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/clients/tests/CMakeLists.txt b/clients/tests/CMakeLists.txt
+index d0b95b4..0339daa 100644
+--- a/clients/tests/CMakeLists.txt
++++ b/clients/tests/CMakeLists.txt
+@@ -64,6 +64,10 @@ if( NOT hiprand_FOUND )
+   find_package( hiprand REQUIRED )
+ endif()
+ 
++if( NOT rocrand_FOUND )
++  find_package( rocrand REQUIRED )
++endif()
++
+ include( ROCMInstallTargets )
+ 
+ set( rocfft-test_source
+@@ -109,6 +113,7 @@ option( BUILD_FFTW "Download and build FFTW" OFF )
+ # look for installed FFTW if we weren't asked to build it
+ if( NOT BUILD_FFTW )
+   find_package( FFTW 3.0 MODULE COMPONENTS FLOAT DOUBLE )
++  set( FFTW_INCLUDES ${FFTW_INCLUDE_DIRS} )
+ endif()
+ 
+ include( ExternalProject )
+@@ -174,6 +179,7 @@ endif()
+ 
+ set( rocfft-test_include_dirs
+   $<BUILD_INTERFACE:${Boost_INCLUDE_DIRS}>
++  $<BUILD_INTERFACE:${rocrand_INCLUDE_DIRS}>
+   $<BUILD_INTERFACE:${FFTW_INCLUDES}>
+   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/misc/include>
+   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../library/src/include>
+@@ -216,6 +222,7 @@ target_link_libraries( rocfft-test
+   hip::device
+   roc::rocfft
+   hip::hiprand
++  roc::rocrand
```

Comment:
If this is required, there's something wrong in the hiprand project. Should file a bug upstream.

Reply:
this is what is being observed -
1 error generated when compiling for gfx1030.
make[2]: *** [clients/tests/CMakeFiles/rocfft-test.dir/build.make:163: clients/tests/CMakeFiles/rocfft-test.dir/accuracy_test_adhoc.cpp.o] Error 1
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/rocfft_accuracy_test.cpp:28:
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/rocfft_accuracy_test.h:24:
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/../../shared/rocfft_params.h:24:
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/../../shared/../shared/fft_params.h:40:
In file included from /tmp/root/spack-stage/spack-stage-rocfft-6.0.2-nlcvyhjfzfumwek5zhdcoq5eri3grfk2/spack-src/clients/tests/../../shared/../shared/data_gen_device.h:36:
In file included from /home/spack/opt/spack/linux-rhel9-skylake_avx512/gcc-11.3.1/hiprand-6.0.2-7mzvsh23dtlwgwzkgqgn2vtgshgkolst/include/hiprand/hiprand.h:66:
/home/spack/opt/spack/linux-rhel9-skylake_avx512/gcc-11.3.1/hiprand-6.0.2-7mzvsh23dtlwgwzkgqgn2vtgshgkolst/include/hiprand/hiprand_rocm.h:24:10: fatal error: 'rocrand/rocrand.h' file not found
#include <rocrand/rocrand.h>
         ^~~~~~~~~~~~~~~~~~~
1 error generated when compiling for gfx1030.
make[2]: *** [clients/tests/CMakeFiles/rocfft-test.dir/build.make:93: clients/tests/CMakeFiles/rocfft-test.dir/rocfft_accuracy_test.cpp.o] Error 1
1 error generated when compiling for gfx1030.
make[2]: *** [clients/tests/CMakeFiles/rocfft-test.dir/build.make:79: clients/tests/CMakeFiles/rocfft-test.dir/gtest_main.cpp.o] Error 1

Reply:
I filed a bug. https://github.com/ROCm/hipRAND/issues/113

Reply:
Do you have an estimate as to when the bug will be fixed upstream? Was wondering if we can wait a bit and retrieve the patch from upstream directly.

Reply:
@alalazo. It has been merged for ROCm 6.3. https://github.com/ROCm/hipRAND/commit/a471c3512ed5d16685d42a89fe43f90458abfba0